### PR TITLE
[Enhancement] Add type check for parquet/csv outfile at FE

### PR DESCRIPTION
**Pull request by letian-jiang**
## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes:

Fixes #

## Problem Summary(Required):

For query `select into outfile`, this PR add supported types check for both parquet/csv outfile. 
If not supported, throws exception at analyze stage.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 13 / 48 (27.08%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/OutFileClause.java|12|47|25.53%|[154, 162, 168, 173, 174, 175, 177, 183, 197, 211, 217, 223, 230, 238, 239, 242, 248, 249, 251, 266, 268, 273, 274, 275, 278, 279, 280, 283, 284, 285, 286, 287, 289, 290, 294]|
|:large_blue_circle:|com/starrocks/sql/analyzer/QueryAnalyzer.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 18 / 59 (30.51%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/OutFileClause.java|17|58|29.31%|[195, 203, 209, 214, 215, 216, 218, 224, 238, 252, 258, 264, 271, 279, 280, 283, 289, 290, 291, 295, 296, 297, 300, 301, 302, 305, 306, 307, 308, 309, 312, 316, 326, 327, 330, 331, 332, 335, 336, 337, 341]|
|:large_blue_circle:|com/starrocks/sql/analyzer/QueryAnalyzer.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/analysis/OutFileClause.java [hunk 1]
```diff
@@ -69,6 +82,45 @@ public class OutFileClause implements ParseNode {
...
+        PrimitiveType.DATE,
+        PrimitiveType.DATETIME,
+        PrimitiveType.CHAR,
+        PrimitiveType.VARCHAR,
+        PrimitiveType.DECIMAL32,
+        PrimitiveType.DECIMAL64,
+        PrimitiveType.DECIMAL128
+    );
+
+    public static final Set<PrimitiveType> CSV_SUPPORTED_PRIMITIVE_TYPES = ImmutableSet.of(
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/analysis/OutFileClause.java:102]: Have you test if each type is supported

**Comment by stephen-shelby** [comment]
Please describe this enhancement and explain why you did it? is there some issues? if yes, please link related issues.

File: fe/fe-core/src/main/java/com/starrocks/analysis/OutFileClause.java (hunk 1 above)
Comment (letian-jiang) [reply fe/fe-core/src/main/java/com/starrocks/analysis/OutFileClause.java:102]: CSV supported types are listed in `csv/converter.cpp`

File: fe/fe-core/src/main/java/com/starrocks/analysis/OutFileClause.java [hunk 2]
```diff
@@ -138,78 +190,155 @@ public BrokerDesc getBrokerDesc() {
...
+        }
+
+        // handle complex type
+        if (type.isStructType()) {
+            return false;
+        }
+
+        if (type.isArrayType()) {
+            ArrayType arrayType = (ArrayType) type;
+            return isSupportedTypeForParquetOutFile(arrayType.getItemType());
```
Comment (Smith-Cruise) [inline fe/fe-core/src/main/java/com/starrocks/analysis/OutFileClause.java:332]: `parquet` ?

File: fe/fe-core/src/main/java/com/starrocks/analysis/OutFileClause.java [hunk 3]
```diff
@@ -138,78 +190,155 @@ public BrokerDesc getBrokerDesc() {
...
+        }
+
+        if (type.isMapType()) {
+            MapType mapType = (MapType) type;
+            return isSupportedTypeForParquetOutFile(mapType.getKeyType()) && isSupportedTypeForParquetOutFile(mapType.getValueType());
```
Comment (Smith-Cruise) [inline fe/fe-core/src/main/java/com/starrocks/analysis/OutFileClause.java:337]: ditto

Comment (Smith-Cruise) [reply fe/fe-core/src/main/java/com/starrocks/analysis/OutFileClause.java:337]: And why not return false directly?

**Comment by Smith-Cruise** [comment]
Test sql `select map_values(col) from tbl outfile xxxxx`.

**Comment by Smith-Cruise** [comment]
Change title: `[Enhancement] Add type check for parquet/csv outfile at FE`

File: fe/fe-core/src/main/java/com/starrocks/analysis/OutFileClause.java (hunk 2 above)
Comment (letian-jiang) [reply fe/fe-core/src/main/java/com/starrocks/analysis/OutFileClause.java]: done

**Review by Smith-Cruise — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 18 / 59 (30.51%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/OutFileClause.java|17|58|29.31%|[195, 203, 209, 214, 215, 216, 218, 224, 238, 252, 258, 264, 271, 279, 280, 283, 289, 290, 291, 295, 296, 297, 300, 301, 302, 305, 306, 307, 308, 309, 312, 316, 326, 327, 330, 331, 332, 335, 336, 337, 341]|
|:large_blue_circle:|com/starrocks/sql/analyzer/QueryAnalyzer.java|1|1|100.00%|[]|

**Review by Youngwb — DISMISSED** [review]

**Review by stephen-shelby — DISMISSED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Pull request merged**